code/modules/mob/living/living_powers.dm

```diff
@@ -32,10 +32,34 @@
 
 	attempt_devour(L, eat_types, mouth_size)
 
-/*
-/mob/living/verb/devourverb(var/mob/living/victim)//For situations where species inherent verbs isnt suitable
-	set category = "Abilities"
-	set name = "Devour Creature"
-	set desc = "Attempt to eat a nearby creature, swallowing it whole if small enough, or eating it piece by piece otherwise"
-	attempt_devour(victim, eat_types, mouth_size)
-*/
\ No newline at end of file
+/mob/living/verb/set_walk_speed()
+	set category = "IC"
+	set name = "Adjust walk speed"
+	set desc = "Allows you to adjust your walking speed to a slower value than normal, or reset it. Does not make you faster."
+
+
+	//First a quick block of code to calculate our normal/best movedelay
+	var/delay = config.walk_speed + movement_delay()
+	var/speed = 1 / (delay/10)
+	var/newspeed
+	var/list/options = list("No limit",speed*0.95,speed*0.9,speed*0.85,speed*0.8,speed*0.7,speed*0.6,speed*0.5, "Custom")
+
+
+	var/response = input(src, "Your current walking speed is [speed] tiles per second. This menu allows you to limit it to a lower value, by applying a multiplier to that. Please choose a value, select custom to enter your own, or No limit to set your walk speed to the maximum. This menu will not make you move any faster than usual, it is only for allowing you to move at a slower pace than normal, for roleplay purposes. Values set here will not affect your sprinting speed", "Limit Walking speed") as null|anything in options
+	if (response == "No limit")
+		src << "Movement speed has now been set to normal, limits removed."
+		src.min_walk_delay = 0
+		return
+	else if (response == "Custom")
+		response = input(src, "Please enter the exact speed you want to walk at, in tiles per second. This value must be less than [speed] and greater than zero", "Limit Walking speed") as num
+		newspeed = response
+	else
```

Comment:
This will create issues.

`null|anything` means that the selection can be exited out of with "Cancel", which gives response a value of `null`. Attempting to convert that into a number and then using it in later operations will bring up issues.

So you should have a catch for it:
```
else if (isnull(response))
    return
```

Alternatively, do this:
```
else if (isnum(response))
    newspeed = text2num(response)
else
    return
```

May want to test, but I _think_ DM recognizes a string that only contains digits as a number. I think.